src/main/java/seedu/duke/helper/Parser.java

```diff
@@ -109,14 +100,14 @@ public static Command parseEditMedicine(String parameters) throws HalpmiExceptio
         return new EditMedicineCommand(medicineParameters);
     }
 
-    public static String[] parseAddAppointment(String parameters) throws HalpmiException {
-        String[] addAppointmentParameters = minParameterCheck(parameters, 6);
+    public static Command parseAddAppointment(String parameters) throws HalpmiException {
+        String[] addAppointmentParameters = minParameterCheck(parameters, 7);
         Validator.validateAddAppointment(addAppointmentParameters);
-        return addAppointmentParameters;
+        return new AddAppointmentCommand(addAppointmentParameters);
     }
 
-    public static String[] parseFindAppointment(String parameters) throws HalpmiException {
-        String[] findAppointmentParameters = minParameterCheck(parameters, 2);
-        return findAppointmentParameters;
+    public static Command parseViewAppointment(String parameters) throws HalpmiException {
```

Comment:
what happens when parameter is null?? eg view appointments

Reply:
It's will throw the HalpMiException stating there is parameters missing